jakartams/src/main/scala/org/apache/pekko/stream/connectors/jakartams/ConnectionRetrySettings.scala

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * license agreements; and to You under the Apache License, version 2.0:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * This file is part of the Apache Pekko project, which was derived from Akka.
+ */
+
+/*
+ * Copyright (C) since 2016 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package org.apache.pekko.stream.connectors.jakartams
+
+import com.typesafe.config.Config
+import org.apache.pekko
+import org.apache.pekko.actor.{ ActorSystem, ClassicActorSystemProvider }
+import org.apache.pekko.util.JavaDurationConverters._
+
+import scala.concurrent.duration._
+
+/**
+ * When a connection to a broker cannot be established and errors out, or is timing out being established or
+ * started, the connection can be retried.
+ * All JMS publishers, consumers, and browsers are configured with connection retry settings.
+ */
+final class ConnectionRetrySettings private (
+    val connectTimeout: scala.concurrent.duration.FiniteDuration,
+    val initialRetry: scala.concurrent.duration.FiniteDuration,
+    val backoffFactor: Double,
+    val maxBackoff: scala.concurrent.duration.FiniteDuration,
+    val maxRetries: Int) {
+
+  /** Time allowed to establish and start a connection. */
+  def withConnectTimeout(value: scala.concurrent.duration.FiniteDuration): ConnectionRetrySettings =
+    copy(connectTimeout = value)
+
+  /** Java API: Time allowed to establish and start a connection. */
+  def withConnectTimeout(value: java.time.Duration): ConnectionRetrySettings = copy(connectTimeout = value.asScala)
+
+  /** Wait time before retrying the first time. */
+  def withInitialRetry(value: scala.concurrent.duration.FiniteDuration): ConnectionRetrySettings =
+    copy(initialRetry = value)
+
+  /** Java API: Wait time before retrying the first time. */
+  def withInitialRetry(value: java.time.Duration): ConnectionRetrySettings = copy(initialRetry = value.asScala)
+
+  /** Back-off factor for subsequent retries. */
+  def withBackoffFactor(value: Double): ConnectionRetrySettings = copy(backoffFactor = value)
```

Comment:
It's a Double? In conf, it seems a Long

Reply:
As a factor it can be fractional (e.g.: https://github.com/apache/pekko-connectors/blob/main/jms/src/main/resources/reference.conf#L73)